### PR TITLE
Funding integration

### PR DESCRIPTION
**Pull request by slint**
- requires https://github.com/inveniosoftware/invenio-rdm-records/pull/936

https://user-images.githubusercontent.com/6756943/167415201-4a87fef7-19bc-47e1-a745-bc8936d26b5b.mov

review requested to @jennur and @kpsherva for UI/UX, if there are tiny issues maybe they can be addressed separately to allow this feature going into the release

### CODE REVIEW

**Comment by ntarocco** [comment]
Can we close https://github.com/inveniosoftware/invenio-app-rdm/pull/768 with this one?

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/detail.html [hunk 1]
```diff
@@ -78,10 +78,44 @@ <h2>{{add_description.type.title_l10n }} <span class="text-muted language">{{ '(
...
+
+          {%- if item.award.number -%}
+            <div class="ui mini basic label" id="number-label-{{ index }}">
+              {{ item.award.number }}
+            </div>
+          {%- endif -%}
+
+          {%- if item.award.identifiers -%}
+            {% for identifier in item.award.identifiers if 'url' == identifier.scheme %}
+              <a href="{{ identifier.identifier }}" target="_blank" rel="noopener noreferrer" aria-label="Open external link">
```
Comment (jennur) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/detail.html:104]:
```suggestion
              <a href="{{ identifier.identifier }}" target="_blank" rel="noopener noreferrer" aria-label="{{_('Open external link')}}">
```

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js [hunk 2]
```diff
@@ -311,18 +267,88 @@ export class RDMDepositForm extends Component {
                 <VersionField />
                 <PublisherField />
               </AccordionField>
-              {/**TODO: uncomment to use FundingField*/}
-              {/* <AccordionField
-              active={true}
-              label={"Funding"}
+
+              <AccordionField
+                fieldPath=""
```
Comment (jennur) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:272]:
```suggestion
                includesPaths={["metadata.funding"]}
```

**Review by jennur — APPROVED** [review]

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js [hunk 3]
```diff
@@ -311,18 +267,88 @@ export class RDMDepositForm extends Component {
...
+                      layout: "list",
+                      page: 1,
+                      size: 5,
+                    },
+                  }}
+                  label="Awards"
+                  labelIcon="money bill alternate outline"
+                  deserializeAward={(award) => {
+                    return {
+                      title: award.title.en ?? award.title,
```
Comment (kpsherva) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:303]: does it make sense to specify separate deserializer function? at least for readability

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js [hunk 4]
```diff
@@ -311,18 +267,88 @@ export class RDMDepositForm extends Component {
...
+                      id: funder.id,
+                      name: funder.name,
+                      ...(funder.pid && { pid: funder.pid }),
+                      ...(funder.country && { country: funder.country }),
+                      ...(funder.identifiers && {
+                        identifiers: funder.identifiers,
+                      }),
+                    };
+                  }}
+                  computeFundingContents={(funding) => {
```
Comment (kpsherva) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:325]: I find this function quite hard to read, would it be possible to simplify ?

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js (hunk 3 above)
Comment (ppanero) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:303]: I can do this. However, my main reason to leave it where it is now was that defining a function would make the code of said function go around line 90 then pass it to the component in line 300, which does not help readability at all (my biased opinion) and in the end it is "just" defining a dict to be returned.

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js (hunk 4 above)
Comment (ppanero) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:325]: Trying... but unsure, the logic is pretty much specific

Comment (ppanero) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:325]: re-worked, I think is more clear now :)

**Comment by ppanero** [comment]
Video updated after applying changes from review

https://user-images.githubusercontent.com/6756943/167575388-63f22f1e-dd38-492a-960b-4ee9013003da.mov

**Pull request merged**